File: crates/ecp-cli/tests/fixtures/incremental_parity/swift/StringUtils.swift
import Foundation

func slugify(_ text: String) -> String {
    let lower = text.lowercased()
    let slug = lower.components(separatedBy: CharacterSet.alphanumerics.inverted)
        .filter { !$0.isEmpty }
        .joined(separator: "-")
    return slug
}

func truncate(_ text: String, maxLen: Int) -> String {
    guard text.count > maxLen else { return text }
    let index = text.index(text.startIndex, offsetBy: maxLen)
    return String(text[..<index]) + "..."
}

func isValidEmail(_ email: String) -> Bool {
    email.contains("@") && email.contains(".")
}

File: crates/ecp-cli/tests/fixtures/incremental_parity/php/StringUtils.php
<?php

namespace App\Utils;

class StringUtils
{
    public static function slugify(string $text): string
    {
        $lower = strtolower($text);
        $slug = preg_replace('/[^a-z0-9]+/', '-', $lower);
        return trim($slug, '-');
    }

    public static function isValidEmail(string $email): bool
    {
        return filter_var($email, FILTER_VALIDATE_EMAIL) !== false;
    }

    public static function truncate(string $text, int $maxLen): string
    {
        return strlen($text) <= $maxLen ? $text : substr($text, 0, $maxLen) . '...';
    }
}
